README.md

```diff
@@ -475,6 +476,58 @@ var err = warden.IsAllowed(&ladon.Request{
 }
 ```
 
+
+##### [Resource Contains Condition](condition_resource_contains.go)
+
+Checks if the string value passed in the access request's context is present in the resource string.
+
+The Condition requires a value string and delimiter (needs to match the resource string) to be passed.
+
+A resource could for instance be: `myrn:some.domain.com:resource:123` and `myrn:some.otherdomain.com:resource:123` (the `:` is then considered a delimiter, and used by the condition to be able to separate the resource components from each other) to allow an action to the resources on `myrn:some.otherdomain.com` you could for instance create a resource condition  with
```

Comment:
I would explain here that the part of the resource "path" must exactly match what you're searching for. So `foo:bar` matches `foo:bar` but not `foo:bara` nor `foo:bara:baz`

Reply:
Todo resource "matches" you only need to input the "full string" in contains, so it makes no use splitting in two different cases (assuming the cost of regex is neglible as you mentioned)...

Not sure how it would work with a resource definition <[^:]+>:foo:bar:<[^:]+>. I always saw resource as an object (singular). 

Can you use regex in the middle of the resource definition like that currently?

`rn:myorg:group:*:device:*`

How would you get device:22 and not device:222 using regex?




Here is an example of the resource condition in Warden:
```
curl -X POST \
  http://localhost:8080/policies \
  -H 'Cache-Control: no-cache' \
  -H 'Content-Type: application/json' \
  -H 'Postman-Token: d279f81a-f1a0-d341-cfea-2816607c00a0' \
  -d '        {
          "description": "One policy to rule them all.",
          "subjects": ["users:peter"],
          "actions" : ["delete"],
          "effect": "allow",
          "resources": [
            "resources:tenant:2"
          ],
          "conditions": {
            "resourceFilter": {
                "type": "ResourceContainsCondition"
            }
          }
        }'
```

```
curl -X POST \
  http://localhost:8080/warden/allowed \
  -H 'Cache-Control: no-cache' \
  -H 'Content-Type: application/javascript' \
  -H 'Postman-Token: 590e4962-c1e0-93dc-3aaf-2133607d6cc9' \
  -d '        {
          "subject": "users:peter",
          "action" : "delete",
          "resource": "resources:tenant:2",
          "context": {
            "resourceFilter": {
            	"Value": "tenant:2"
            	
            }
          }
        }'
```

Reply:
> Can you use regex in the middle of the resource definition like that currently?
>
> rn:myorg:group:*:device:*

`rn:myorg:group:<.*>:device:<.*>`

> How would you get device:22 and not device:222 using regex?

I can't give you a full tutorial on regex, but doing stuff like `device:22[^\:]+` is the way to go.

In your example, I don't really get the need for checking for `tenant:2` as the policy itself is already doing that (it must match `"resources:tenant:2"`). Why do you need a second thing (condition) to solve what the resource part of the policy definition itself is already capable of? Maybe you could explain your use case?

Reply:
please refer to my first comment on this pr.

Reply:
Ah sorry, the discussion has been going for a while that I forgot about it :) So instead of having tons of policies reflecting that, you simply want to do that using the context - that makes sense IMO.

The main issue with the current condition is that you're assuming that we have delimiters while in fact this might not be the case for some people. So instead it would be more useful to match this in total with a regular expression (leave the delimiter out completely). The example above would be covered with `(^|:.*)tenant:2($|:.*)` (this matches `tenant:2`, `tenant:2:123`, `123:tenant:2:123` but not `tenant:22`). Obviously, as you know the layout of your resource names you probably won't even need that and simply checking for `:tenant:2:` or `resources:tenant:2` would already be enough (no regexp required).

I suggest to throw out the delimiter part and either have one or two conditions instead:

* ResourceContains (uses strings.Contains())
* ResourceMatches (uses regexp.MatchString())